socs/agents/generator/agent.py

```diff
@@ -0,0 +1,361 @@
+        for string in config:
+            self.read_strings.append(ReadString(string))
+
+    def read_cycle(self, client):
+        for i, val in enumerate(self.read_strings):
+            data = self.read_regs(client, val)
+            return data
+
+    def read_regs(self, client, register_string_obj):
+        try:
+            data = register_string_obj.read(client)
+        except Exception as e:
+            print('error in read', e)
+            return
+
+        return data
+
+    def init_generator(self, session, params=None):
+        """init_generator(auto_acquire=False)
+
+        **Task** - Perform first time setup of the Generator.
+
+        Parameters:
+            auto_acquire (bool, optional): Starts data acquisition after
+                initialization if True. Defaults to False.
+
+        """
+        if params is None:
+            params = {}
+
+        auto_acquire = params.get('auto_acquire', False)
+
+        if self.initialized:
+            return True, "Already initialized."
+
+        with self.lock.acquire_timeout(3, job='init') as acquired:
+            if not acquired:
+                self.log.warn("Could not start init because "
+                              "{} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('starting')
+
+            client = ModbusClient(host=self.host, port=self.port, auto_open=True, auto_close=False)
+            if client.open():
+                self.client = client
+                self.initialized = True
+            else:
+                self.initialized = False
+                return False, 'Could not connect to generator'
+
+        # Start data acquisition if requested
+        if auto_acquire:
+            self.agent.start('acq')
+
+        return True, 'Generator initialized.'
+
+    def acq(self, session, params=None):
+        """acq()
+
+        **Process** - Start data acquisition.
+
+        Notes
+        -----
+        The most recent data collected is stored in session data in the
+        structure::
+
+            >>> response.session['data']
+            {"fields":
+                {'Oil_pressure': {'value': 100.0, 'units': 'Kpa'},
+                 'Coolant_temperature': {'value': 20.0, 'units': 'Degrees C'},
+                 'Oil_temperature': {'value': 20.0, 'units': 'Degrees C'},
+                 'Fuel_level': {'value': 100.0, 'units': '%'},
+                 'Charge_alternator_voltage': {'value': 10.0, 'units': 'V'},
+                 'Engine_Battery_voltage': {'value': 10.0, 'units': 'V'},
+                 'Engine_speed': {'value': 4000, 'units': 'RPM'},
+                 'Generator_frequency': {'value': 1.0, 'units': 'Hz'},
+                 ...
+                 'connection': {'last_attempt': 1680812613.939653, 'connected': True}},
+             "timestamp":1601925677.6914878}
+
+        Refer to the config file for all possible fields and their respective
```

Comment:
We should maybe link to or insert the config file, since we're packaging it, and users probably don't know where to find it. (Or put every register in this example.)